### ISSUE TITLE
Does not accept forward reference to type alias

### ISSUE THREAD

**Issue by fluggo** [microsoft/pyright#458]
Pyright does not understand a forward reference to a type alias, but does accept backwards ones.

```python
from typing import Dict, Any, Union

class Base:
    @staticmethod
    def create(data: Dict[str, Any]) -> 'Mix':
        return A()

class A(Base):
    pass

class B(Base):
    pass

Mix = Union[A, B]

class S:
    @staticmethod
    def create(data: Dict[str, Any]) -> 'Mix':
        return A()
```

In this code, pyright flags the `'Mix'` reference for the return type of Base.create() with `'Mix' is unbound`, but the return type of S.create() is fine.

![Screen Shot 2019-12-27 at 10 14 31 AM](https://user-images.githubusercontent.com/2552507/71524145-b9c0a300-2891-11ea-9702-84b8a73e2755.png)

**Expected behavior**
Pyright accepts `Union['A', 'B']` as a forward reference and accepts `'Mix'` as a backward reference, so it should accept `'Mix'` as a forward reference.

**VS Code extension or command-line**
This is ms-pyright 1.1.14 in VS Code on Python 3.8.0.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.1.16, which I just published.

**Comment by fluggo** [comment]
Success! Thanks much.
